src/transfers-history/model/state.ts

```diff
@@ -0,0 +1,102 @@
+import { BigNumberish } from "ethers";
+import { ChainId } from "../../constants";
+
+export type TransferStatus = "pending" | "filled";
+
+export type Transfer = {
+  depositId: number;
+  depositTime: number;
+  status: TransferStatus;
+  filled: BigNumberish;
+  sourceChainId: ChainId;
+  destinationChainId: number;
+  assetAddr: string;
+  amount: BigNumberish;
+  depositTxHash: string;
+};
+
+export type TransferFilters = {
+  address?: string;
+  status: TransferStatus;
+};
+
+type QueryProgress = { latestFromBlock?: number; latestToBlock?: number; latestFromTimestamp?: number };
+type ChainQueryProgress = Record<number, QueryProgress>;
+
+type ChainTransfers = {
+  [chainId: number]: {
+    [depositId: number]: Transfer;
+  };
+};
+
+type BlockLowerBound = {
+  [chainId: number]: number;
+};
+
+const blockLowerBoundInitialValues = {
+  [ChainId.MAINNET]: 0,
+  [ChainId.RINKEBY]: 0,
+  [ChainId.KOVAN]: 0,
+  [ChainId.OPTIMISM_KOVAN]: 0,
+  [ChainId.ARBITRUM_RINKEBY]: 9828565,
+};
+
+export class State {
```

Comment:
I think this name is a little nonspecific. What is this state tracking? What is it meant to be used for?

Reply:
I completely change the implementation here. I'm using the Repository pattern now